rust/ffi/node/src/index/vector.rs

```diff
@@ -51,66 +56,43 @@ pub(crate) fn table_create_vector_index(mut cx: FunctionContext) -> JsResult<JsP
-                index_builder.metric_type(metric_type);
-            }
-
-            let num_partitions = obj.get_opt_usize(cx, "num_partitions")?;
-            let max_iters = obj.get_opt_usize(cx, "max_iters")?;
-
-            num_partitions.map(|np| {
-                let max_iters = max_iters.unwrap_or(50);
-                let ivf_params = IvfBuildParams {
-                    num_partitions: np,
-                    max_iters,
-                    ..Default::default()
-                };
-                index_builder.ivf_params(ivf_params)
-            });
-
-            if let Some(use_opq) = obj.get_opt::<JsBoolean, _, _>(cx, "use_opq")? {
-                pq_params.use_opq = use_opq.value(cx);
-            }
-
-            if let Some(num_sub_vectors) = obj.get_opt_usize(cx, "num_sub_vectors")? {
-                pq_params.num_sub_vectors = num_sub_vectors;
-            }
-
-            if let Some(num_bits) = obj.get_opt_usize(cx, "num_bits")? {
-                pq_params.num_bits = num_bits;
-            }
-
-            if let Some(max_iters) = obj.get_opt_usize(cx, "max_iters")? {
-                pq_params.max_iters = max_iters;
-            }
+    builder: &mut IndexBuilder,
+) -> crate::error::Result<()> {
+    match obj.get::<JsString, _, _>(cx, "type")?.value(cx).as_str() {
+        "ivf_pq" => builder.ivf_pq(),
+        _ => {
+            return Err(InvalidIndexType {
+                index_type: "".into(),
+            })
+        }
+    };
 
-            if let Some(max_opq_iters) = obj.get_opt_usize(cx, "max_opq_iters")? {
-                pq_params.max_opq_iters = max_opq_iters;
-            }
+    obj.get_opt::<JsString, _, _>(cx, "index_name")?
+        .map(|s| builder.name(s.value(cx).as_str()));
 
-            if let Some(replace) = obj.get_opt::<JsBoolean, _, _>(cx, "replace")? {
-                index_builder.replace(replace.value(cx));
-            }
+    if let Some(metric_type) = obj.get_opt::<JsString, _, _>(cx, "metric_type")? {
+        let metric_type = MetricType::try_from(metric_type.value(cx).as_str())?;
+        builder.metric_type(metric_type);
+    }
 
-            Ok(index_builder)
+    if let Some(np) = obj.get_opt_usize(cx, "num_partitions")? {
+        builder.num_partitions(np as u64);
+    }
+    if let Some(ns) = obj.get_opt_u32(cx, "num_sub_vectors")? {
+        builder.num_sub_vectors(ns);
+    }
+    if let Some(max_iters) = obj.get_opt_u32(cx, "max_iters")? {
+        builder.max_iterations(max_iters);
+    }
+    if let Some(num_bits) = obj.get_opt_u32(cx, "num_bits")? {
+        builder.num_bits(num_bits);
+    }
+    if let Some(replace) = obj.get_opt::<JsBoolean, _, _>(cx, "replace")? {
+        if replace.value(cx) {
+            builder.replace();
         }
-        index_type => Err(InvalidIndexType {
-            index_type: index_type.into(),
-        }),
+    } else {
+        builder.replace();
     }
```

Comment:
Very minor nit:  If you want the default behavior to be `replace=true` then maybe just make the default for `IndexBuilder` to be `replace=true` and then change `builder::replace` to `builder::dont_replace` (or change `builder::replace` to take a boolean arg)

Reply:
Sounds good.